consensus/cometbft/service/abci.go

```diff
@@ -36,11 +36,29 @@ var (
 	errNilFinalizeBlockState = errors.New("finalizeBlockState is nil")
 )
 
+// combineContexts combines the CometBFT context with the original context from the node.
+// TODO: This can be removed if CometBFT implements contexts.
+func (s *Service) combineContexts(ctx context.Context) context.Context {
+	if s.ctx == nil {
+		return ctx
+	}
+	newCtx, cancel := context.WithCancel(context.Background())
```

Comment:
why not just use a parent context here? Seems like the perfect use case for this secenario